src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2904,19 +2916,71 @@ private SocketFactory getSocketFactory() throws IOException {
         return socketFactory;
     }
 
-    // This method contains the old logic of connecting to
-    // a socket of one of the IPs corresponding to a given host name.
-    // In the old code below, the logic around 0 timeout has been removed as
-    // 0 timeout is not allowed. The code has been re-factored so that the logic
-    // is common for hostName or InetAddress.
-    private Socket getDefaultSocket(String hostName, int portNumber, int timeoutInMilliSeconds) throws IOException {
-        // Open the socket, with or without a timeout, throwing an UnknownHostException
-        // if there is a failure to resolve the host name to an InetSocketAddress.
-        //
+    /**
+     * This method makes a connected socket given a hostname or IP address.
+     * 
+     * @param hostName
+     *        Hostname or IP Address
+     * @param portNumber
+     *        Port number
+     * @param timeoutInMilliSeconds
+     *        Timeout
+     * @param IPAddressPreference
+     *        IPv4First (default), IPv6First or UsePlatformDefault
+     * @return Connected Socket
+     * @throws IOException
+     * @throws SQLServerException
+     */
+    private Socket getSocketByIPvPreference(String hostName, int portNumber, int timeoutInMilliSeconds,
```

Comment:
why do we need to change the name of this method just because 1 param has been added? This method gets the default (ie not custom) socket by hostname, portnum timeout and ipPreference. I think it's better to keep the orig name as it's more descriptive

Reply:
I think the behavior has changed enough that "default" is not descriptive enough. getDefaultSocket only ever did one thing regardless of inputs while this function will change order it gets prioritizes sockets based on parameters.

Reply:
Should no longer be any IP'v' s anywhere